src/routes/maven.rs

```diff
@@ -0,0 +1,262 @@
+            .await
+            .map_err(|e| ApiError::DatabaseError(e.into()))?;
+    }
+
+    let user_option = get_user_from_headers(req.headers(), &**pool).await.ok();
+
+    if let Some(data) = mod_data {
+        let mut authorized = !data.status.is_hidden();
+
+        if let Some(user) = user_option {
+            if !authorized {
+                if user.role.is_mod() {
+                    authorized = true;
+                } else {
+                    let user_id: database::models::ids::UserId = user.id.into();
+
+                    let mod_exists = sqlx::query!(
+                        "SELECT EXISTS(SELECT 1 FROM team_members WHERE team_id = $1 AND user_id = $2)",
+                        data.inner.team_id as database::models::ids::TeamId,
+                        user_id as database::models::ids::UserId,
+                    )
+                    .fetch_one(&**pool)
+                    .await
+                    .map_err(|e| ApiError::DatabaseError(e.into()))?
+                    .exists;
+
+                    authorized = mod_exists.unwrap_or(false);
+                }
+            }
+        }
+
+        if authorized {
+            let version_names = sqlx::query!(
+                "
+                SELECT version_number, release_channels.channel channel
+                FROM versions
+                LEFT JOIN release_channels ON release_channels.id = versions.release_channel
+                WHERE mod_id = $1
+                ",
+                data.inner.id as database::models::ids::ModId
+            )
+            .fetch_all(&**pool)
+            .await
+            .map_err(|e| ApiError::DatabaseError(e.into()))?;
+
+            let respdata = Metadata {
+                group_id: "maven.modrinth".to_string(),
+                artifact_id: string,
+                versioning: Versioning {
+                    latest: version_names
+                        .last()
+                        .map_or("release", |x| &x.version_number)
+                        .to_string(),
+                    release: version_names
+                        .iter()
+                        .rfind(|x| x.channel == "release")
+                        .map_or("", |x| &x.version_number)
+                        .to_string(),
+                    versions: Versions {
+                        versions: version_names
+                            .iter()
+                            .map(|x| x.version_number.clone())
+                            .collect::<Vec<_>>(),
+                    },
+                    last_updated: data.inner.updated.format("%Y%m%d%H%M%S").to_string(),
+                },
+            };
+
+            // return Ok(HttpResponse::Ok().content_type("text/xml").body(response));
+            Ok(HttpResponse::Ok()
+                .content_type("text/xml")
+                .body(yaserde::ser::to_string(&respdata).map_err(|e| ApiError::XmlError(e))?))
+        } else {
```

Comment:
Rust allows for early returns, so an easy fix for that nesting would be to invert the conditions, and just return the `HttpResponse::NotFound` error.